### ISSUE TITLE
al_save_bitmap with DDS outputs nothing

### ISSUE THREAD

**Issue by katastic** [liballeg/allegro5#1339]
DDS doesn't appear to output any file at all (on Linux) even though the documentation says PCX, BMP, TGA and DDS are always supported.

https://liballeg.org/a5docs/5.2.2/image.html
> The following types are built into the Allegro image addon and guaranteed to be available: BMP, DDS, PCX, TGA. Every platform also supports JPEG and PNG via external dependencies.

Using Linux, and DAllegro.

Perhaps DDS is only supported for **loading** and there's an empty save routine?

**No error or segfault is produced** when attempting to output to DDS

**Comment by katastic** [comment]
Ah, I was expecting an assert for calling a missing piece of code, not a silent return. 
 Fair enough. The docs make no mention of ALLEGRO_ERROR.

Allegro  also apparently doesn't dump to Allegro.log if its is compiled in release mode? What was the rational there? No logging of critical errors... just because it's release mode? How could a few save_bitmap errors, for example, affect performance? Certainly tens-of-thousand-call functions like al_put_pixel aren't dumping to allegro.log is it? Also searching google for ALLEGRO_ERROR or allegro.log on the A.CC man page comes up with nothing.

site:https://www.allegro.cc/manual/5/ allegro.log
site:https://www.allegro.cc/manual/5/ ALLEGRO_ERROR

A search through github for allegro_error uses:

https://github.com/liballeg/allegro5/search?q=allegro_error

shows pretty typical error messages. "Out of memory" for example. Isn't that kind of important even when in release mode, being run on machines other than the dev machine?

**Comment by SiegeLord** [comment]
Yes, `al_save_bitmap` should return false if it doesn't already.

We probably should document the logging setup somewhere. It's disabled in release mode... I guess for historical reasons. It's possible to enable it at runtime (even in release mode, unless disabled at compile time) by doing:

```c
al_set_config_value(al_get_system_config(), "trace", "level", "debug");
```

before `al_init`.

**Comment by katastic** [comment]
> al_set_config_value(al_get_system_config(), "trace", "level", "debug");

This is helpful, thank you. We should definitely document logging--especially since it's an expected part of development--detailed error messages. Many people (like myself) assume that "debug" means "debug symbols and no optimization (plus asserts)" not "doesn't output any error messages at all."

 And I realize DDS saving is probably a super rare scenario (and I don't even need it) but it is apart of the API so it should be documented that it's "load only". 

Thanks,
--Chris

**Comment by EdgarReynaldo** [comment]
Read the example allegro5.cfg file that comes with allegro. It tells you what is what. There is also this : https://github.com/liballeg/allegro_wiki/wiki/Debugging-and-logging-with-A5 . Those D 'debug' warnings are freetype and allegro caching font glyphs. Usually you want to set the trace level to error and let it filter the junk for you.
